google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java

```diff
@@ -108,12 +111,20 @@
   private JButton moduleSettingsButton;
   private JCheckBox hiddenValidationTrigger;
   private JLabel noSupportedModulesWarning;
+  private JLabel environmentLabel;
 
   private static final boolean PROMOTE_DEFAULT = false;
   private static final boolean STOP_PREVIOUS_VERSION_DEFAULT = false;
 
   public AppEngineFlexibleDeploymentEditor(Project project, AppEngineDeployable deploymentSource) {
     this.deploymentSource = deploymentSource;
+
+    gcpProjectSelector = appEngineConfig.getProjectSelector();
```

Comment:
basically whats happening is that the common panel is housing these common UX components. Then the flex and the standard editors may need access to some of these common elements so that they can handle events (and labels etc.) differently. 

In this PR I was just reassigning the already existing members (since everything was duplicated) as you see in this line you referred to. I did this partly to minimize the diff in the PR.

But I think it would be better to not do this reassignment and get the common elements directly from the common panel. If you have a better pattern I'm very open to suggestions.